src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java

```diff
@@ -0,0 +1,360 @@
+	 * @param pixelPath From 0 to half of screen X, if path will be too high it will be set to edge of app
+	 * @param zoomWay Use public const ZOOM_WAY from that class
+	 * @param waitAfter In milliseconds
+	 */
+	public void zoomInOutPointXY(int pointX, int pointY, int fingersSpace, int pixelPath, String zoomWay, int waitAfter) {
+		pointX = (int)((pointX / 100f) * appNativeWidth);
+		pointY = (int)(((pointY / 100f) * appNativeHeight) + taskbarNativeHeight);
+		TouchAction touchOne = new TouchAction(mobileDriver);
+		TouchAction touchTwo = new TouchAction(mobileDriver);
+		MultiTouchAction multiTouch = new MultiTouchAction(mobileDriver);
+		int halfPath = 0;
+		int endXOne = 0;
+		int endXTwo = 0;
+		int offSet = 0;
+		int startXOne = 0;
+		int startXTwo = 0;
+		if (fingersSpace >= pixelPath) {
+			fingersSpace = pixelPath - 2;
+		}
+		fingersSpace /= 2;
+		if (pixelPath < appNativeWidth) {
+			halfPath = pixelPath / 2;
+		} else {
+			halfPath = (appNativeWidth - 2) / 2;
+		}
+		endXOne = pointX - halfPath;
+		if (endXOne < 0) {
+			offSet = -endXOne;
+			endXOne = 0;
+		}
+		endXTwo = pointX + halfPath;
+		if (endXTwo > (appNativeWidth - 2)) {
+			offSet = (appNativeWidth - 2) - endXTwo;
+			endXTwo = (appNativeWidth - 2);
+		}
+		if (offSet > 0) {
+			endXTwo += offSet;
+		}
+		if (offSet < 0) {
+			endXOne += offSet;
+		}
+		startXOne = pointX - fingersSpace;
+		if (startXOne <= 0) {
+			startXOne = 1;
+		}
+		startXTwo = pointX + fingersSpace;
+		if (startXTwo >= (appNativeWidth - 2)) {
+			startXTwo = appNativeWidth - 3;
+		}
+		if (zoomWay == "out") {
+			int temp;
+			temp = startXOne;
+			startXOne = endXOne;
+			endXOne = temp;
+			temp = startXTwo;
+			startXTwo = endXTwo;
+			endXTwo = temp;
+		}
+		if (mobileDriver.getContext() != "NATIVE_APP") {
+			mobileDriver.context("NATIVE_APP");
+		}
+		touchOne.press(startXOne, pointY).moveTo(endXOne - startXOne, 0).release();
+		touchTwo.press(startXTwo, pointY).moveTo(endXTwo - startXTwo, 0).release();
+		multiTouch.add(touchOne);
+		multiTouch.add(touchTwo);
+		try {
+			multiTouch.perform();
+		} catch (Exception e) {}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
+		if (mobileDriver.getContext() != "WEBVIEW_1") {
+			mobileDriver.context("WEBVIEW_1");
+		}
+	}
+	
+	/**
+	 * 
+	 * @param pointX Use value 0-100, it is percent of app width, Reccomend 10-90
+	 * @param pointY Use value 0-100, it is percent of app height, Reccomend 10-90
+	 * @param duration In milliseconds, Reccomend 500
+	 * @param waitAfter In milliseconds
+	 */
+	public void tapOnPointXY(int pointX, int pointY, int duration, int waitAfter) {
+		pointX = (int)((pointX / 100f) * appNativeWidth);
+		pointY = (int)(((pointY / 100f) * appNativeHeight) + taskbarNativeHeight);
+		if (mobileDriver.getContext() != "NATIVE_APP") {
+			mobileDriver.context("NATIVE_APP");
+		}
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {}
```

Comment:
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.